plugins/performance-lab/includes/site-health/audit-autoloaded-options/helper.php

```diff
@@ -211,7 +211,24 @@ function perflab_aao_get_autoloaded_options_table(): string {
 function perflab_aao_get_disabled_autoloaded_options_table(): string {
 	$disabled_options = get_option( 'perflab_aao_disabled_options', array() );
 
-	if ( ! is_array( $disabled_options ) ) {
+	$autoload_option_names = array_map(
+		static function ( $option ) {
+			return $option->option_name;
+		},
+		perflab_aao_query_autoloaded_options()
+	);
+
+	// Filter out the $disabled_options that are present in $autoload_option_names.
+	$disabled_options = array_filter(
+		$disabled_options,
+		static function ( $option ) use ( $autoload_option_names ) {
+			return ! in_array( $option, $autoload_option_names, true );
+		}
+	);
+
+	update_option( 'perflab_aao_disabled_options', $disabled_options );
```

Comment:
Shouldn't this only do an update if there was a change made to the options? Granted, `update_option()` will only proceed if there is a change with the old value. But it looks a bit strange here to be calling the update in a function that is just getting the table.

Reply:
@westonruter Yes it look strange to update value their, I have tried to update it on `init` hook but it will reflect the value after second time reload. When we click on Disabled Autoload it shows the option in both tables but if we reload it then it removed the value from disabled table. It because of `wp_load_alloptions()` cache?

```
add_action( 'init','perflab_aao_register_init_action' );
function perflab_aao_register_init_action() {
	if ( isset( $_GET['autoload_updated'] ) && 'true' === $_GET['autoload_updated'] ) {
		$disabled_options = (array) get_option( 'perflab_aao_disabled_options', array() );
		$autoload_option_names = array_map(
			static function ( object $option ): string {
				return $option->option_name;
			},
			perflab_aao_query_autoloaded_options()
		);

		// Filter out the $disabled_options that are present in $autoload_option_names.
		$disabled_options = array_filter(
			$disabled_options,
			static function ( $option ) use ( $autoload_option_names ): bool {
				return ! in_array( $option, $autoload_option_names, true );
			}
		);

		update_option( 'perflab_aao_disabled_options', $disabled_options );
	}
}
```

cc. @joemcgill

Reply:
It will works. Thanks!